core/src/appleMain/kotlin/CBPeripheralCoreBluetoothAdvertisement.kt

```diff
@@ -70,14 +70,14 @@ internal class CBPeripheralCoreBluetoothAdvertisement(
     override fun equals(other: Any?): Boolean {
         if (this === other) return true
         if (other !is CoreBluetoothAdvertisement) return false
-        if (identifier.UUIDString != other.identifier.UUIDString) return false
+        if (identifier.toString() != other.identifier.toString()) return false
```

Comment:
`Uuid` has `equals`, so no need to convert to `String`.

```suggestion
        if (identifier != other.identifier) return false
```

Reply:
Do you have some idea??

`Operator '!=' cannot be applied to 'Identifier /* = Uuid */' and 'Identifier'`

Reply:
Bizarre, I'm not seeing that error:

![Screenshot 2023-07-05 at 10 39 19 AM](https://github.com/JuulLabs/kable/assets/98017/94f9e1a4-7bf6-4bef-ab26-fa6f79020c34)

Maybe we need to re-Gradle sync the project? 🤷 

Can you try making the change and just seeing if it compiles anyways (even with the error in the IDE)?

Reply:
The error is just in IDE. What android version are you using?

Reply:
![Screenshot 2023-07-05 at 12 34 31 PM](https://github.com/JuulLabs/kable/assets/98017/ff502af5-fae2-48cc-85d8-6fd5d1362f7b)